src/javascripts/Services/hotkeyService.js

```diff
@@ -33,19 +33,30 @@ ngapp.service('hotkeyService', function(hotkeyFactory) {
             if (!action) return;
             action = action.callback;
         }
-        action.Constructor === Function ? action(scope, e) : scope[action](e);
+        typeStr === 'function' ? action(scope, e) : scope[action](e);
         e.stopImmediatePropagation();
         e.preventDefault();
     };
 
-    this.buildOnKeyDown = function(scope, label, view) {
-        let hotkeys = hotkeyFactory[`${view}Hotkeys`];
-        scope[label] = function(e) {
+    let keyEventHandler = function(scope,hotkeys,type) {
+        return function (e) {
+            if (e.type !== type)
```

Comment:
let's make this a one-line if statement to match the style used in the rest of the codebase.  so:
```js
if (e.type !== type) return;
```